src/lsstseries/ensemble.py

```diff
@@ -294,14 +364,73 @@ def from_parquet(
         else:
             columns = [self._time_col, self._flux_col, self._err_col, self._band_col]
 
-        # Read in a parquet file
-        self._data = dd.read_parquet(file, index=self._id_col, columns=columns, split_row_groups=True)
+        # Read in the source parquet file(s)
+        self._source = dd.read_parquet(
+            source_file, index=self._id_col, columns=columns, split_row_groups=True
+        )
 
         if npartitions and npartitions > 1:
-            self._data = self._data.repartition(npartitions=npartitions)
+            self._source = self._source.repartition(npartitions=npartitions)
         elif partition_size:
-            self._data = self._data.repartition(partition_size=partition_size)
+            self._source = self._source.repartition(partition_size=partition_size)
```

Comment:
This seems to not be directly connected to this pull, but it seems tricky that user is allowed to specify both partitioning options (npartitions, partition_size) and one of them gets clobbered. Is that what dask does also?

Reply:
Yeah it is a bit tricky, and also what Dask does: https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.repartition.html